llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -437,6 +437,16 @@ std::string saveModuleProperties(module_split::ModuleDesc &MD,
       MetadataNames.push_back(Func.getName().str() + "@reqd_work_group_size");
       ProgramMetadata.insert({MetadataNames.back(), KernelReqdWorkGroupSize});
     }
+
+    for (auto &GV : M.globals()) {
+      if (!isDeviceGlobalVariable(GV))
+        continue;
+
+      StringRef GlobalID = getGlobalVariableUniqueId(GV);
+      std::vector<char> GVNameVec(GV.getName().begin(), GV.getName().end());
+      MetadataNames.push_back(GlobalID.str() + "@global_id_mapping");
+      ProgramMetadata.insert({MetadataNames.back(), GVNameVec});
```

Comment:
Can we update `PropertyValue` constructors to accept `StringRef` directly there? The thing is, we will anyway re-allocate memory and copy input data within `PropertyValue`, so here we have intermediate allocation done by `GVNameVec`, which will be thrown away.

Reply:
Good call! This has been done in https://github.com/intel/llvm/pull/7799.